src/components/Cluster/ClusterDetail/AddNodePool.js

```diff
@@ -177,20 +203,46 @@ class AddNodePool extends Component {
         valid: true,
         value: this.props.defaultInstanceType,
       },
+      useAlike: false,
+      instanceDistribution: {
+        onDemandBaseCapacity: 0,
+        spotInstancePercentage: 0,
+      },
     },
     awsInstanceTypes: {},
+    allowSpotInstances: false,
+    allowAlikeInstances: false,
   };
 
   componentDidMount() {
     this.setState({
       awsInstanceTypes: JSON.parse(window.config.awsCapabilitiesJSON),
+      ...this.computeInstanceCapabilities(),
     });
```

Comment:
Do these have to be in state? Do they ever change?

Reply:
On cluster creation, when the user selects another version.